### ISSUE TITLE
Doesn't update in Wordpress

### ISSUE THREAD

**Issue by GoetzMueller** [podlove/podlove-publisher#705]
Since 2 weeks the plugin doesn't update in Wordpress with none of the build-in update methods. Any idea (remember you made a manual change in one or several files of the plugin to fix an issue)?

Website             http://www.geemco.de
PHP Version         5.5.22
WordPress Version   4.1.1
Publisher Version   2.0.2
Web Player Version  2.0.17
curl Version        7.21.0
iconv               available
simplexml           ok
max_execution_time  30
upload_max_filesize 2M
memory_limit        256M
disable_classes     
disable_functions   
permalinks          ok
podcast_settings    
web_player          ok
podlove_cache       on
assets              
	mp3    audio/mpeg
	m4a    audio/mp4

0 errors
0 notices
Nice, Everything looks fine!

**Comment by timpritlove** [comment]
Can you explain what you exactly mean by "update in Wordpress” (what should be updated and where?) and which "build-in update methods" do you refer to? Not sure I am understanding what stopped working here.

**Comment by GoetzMueller** [comment]
I meant the update process for plugins ie. Podlove within the WP backend (instead of uploading the files via ftp).

One way is to do it from the list of installed plugins. It uploads the code, unpacks it but then it stops.

The other way is via the update notification (in the WP top bar). Here it (WP) initially sets the maintenance mode, then proceeds as described above. Which never finishes and I have to reset the maintenance mode manually (by deleting the .maintenance file).

**Comment by GoetzMueller** [comment]
Any news about that one? Any further information from my side necessary?

**Comment by timpritlove** [comment]
I must admit I don't know what is going wrong here. The plugin update mechanism is WP stuff and AFAIK our code is not involved in that process at all. We have hundreds of sites updating the plugin without problems so I assume there is something in your installation that is going wrong.

* Do all other plugins update properly?
* Is there anything in the error log file that can point to the problem here?

**Comment by GoetzMueller** [comment]
I have no issues with updating other plugins. Just did so today with two other plugins.

Which error log file do you mean?

Could the local changes that Eric Teubert made with the plugin code in my installation be of any influence? I hope these changes made in the current version ;-)

Could anything go wrong when I upload the 2.0.5 code manually via FTP? Especially regarding any configuration that has to be adapted which I assume could be done by update specific code within the plugin.

**Comment by timpritlove** [comment]
I am talking about the Web Server's Error Log

I would suspect  there are permission problems. Local changes via FTP upload might have screwed up  file permissions and/or file ownership. Updating the file might not be possible for WP as the server process can't change files. Make sure the plugin directory and its files are writeable by the web server process.

**Comment by GoetzMueller** [comment]
Below are the last 5 minutes of the log, right after an update attempt. Nothing strange as far as I understand it. The file permissions and ownership definition also look "quite" ok. Due to the huge amount of files in the Podlove plugin directory I have not checked all of them.

Is there a way to get Eric in the loop? Maybe he noted which files he changed.

84.57.178.67 - - [04/Apr/2015:22:28:19 +0200] "POST /wp-admin/admin-ajax.php HTTP/1.1" 200 135 www.geemco.de "http://www.geemco.de/wp-admin/post.php?post=4479&action=edit&message=1" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10_2) AppleWebKit/600.4.10 (KHTML, like Gecko) Version/8.0.4 Safari/600.4.10" "-"
84.57.178.67 - - [04/Apr/2015:22:28:50 +0200] "POST /wp-admin/admin-ajax.php HTTP/1.1" 200 98 www.geemco.de "http://www.geemco.de/wp-admin/post.php?post=4479&action=edit&message=1" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10_2) AppleWebKit/600.4.10 (KHTML, like Gecko) Version/8.0.4 Safari/600.4.10" "-"
84.57.178.67 - - [04/Apr/2015:22:28:54 +0200] "GET /wp-admin/index.php HTTP/1.1" 200 110439 www.geemco.de "http://www.geemco.de/wp-admin/post.php?post=4479&action=edit&message=1" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10_2) AppleWebKit/600.4.10 (KHTML, like Gecko) Version/8.0.4 Safari/600.4.10" "-"
84.57.178.67 - - [04/Apr/2015:22:28:55 +0200] "POST /wp-admin/admin-ajax.php HTTP/1.1" 200 1 www.geemco.de "http://www.geemco.de/wp-admin/post.php?post=4479&action=edit&message=1" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10_2) AppleWebKit/600.4.10 (KHTML, like Gecko) Version/8.0.4 Safari/600.4.10" "-"
84.57.178.67 - - [04/Apr/2015:22:28:56 +0200] "GET /wp-content/plugins/disqus-comment-system/media/js/dashboard.js?ver=4.1.1 HTTP/1.1" 200 800 www.geemco.de "http://www.geemco.de/wp-admin/index.php" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10_2) AppleWebKit/600.4.10 (KHTML, like Gecko) Version/8.0.4 Safari/600.4.10" "-"
84.57.178.67 - - [04/Apr/2015:22:28:56 +0200] "GET /wp-admin/load-scripts.php?c=1&load%5B%5D=jquery-core,jquery-migrate,jquery-ui-core,jquery-ui-datepicker,jquery-ui-widget,jquery-ui-mouse,jquery-ui-sortable,utils,json2&ver=4.1.1 HTTP/1.1" 200 64560 www.geemco.de "http://www.geemco.de/wp-admin/index.php" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10_2) AppleWebKit/600.4.10 (KHTML, like Gecko) Version/8.0.4 Safari/600.4.10" "-"
84.57.178.67 - - [04/Apr/2015:22:28:56 +0200] "GET /wp-admin/load-scripts.php?c=1&load%5B%5D=hoverIntent,common,admin-bar,wp-ajax-response,jquery-color,wp-lists,quicktags,jquery-query,admin-comments,postbox,dashboard,unde&load%5B%5D=rscore,customize-base,customize-loader,thickbox,plugin-install,shortcode,media-upload,svg-painter,heartbeat,wp-auth-check,word-c&load%5B%5D=ount,wplink&ver=4.1.1 HTTP/1.1" 200 46894 www.geemco.de "http://www.geemco.de/wp-admin/index.php" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10_2) AppleWebKit/600.4.10 (KHTML, like Gecko) Version/8.0.4 Safari/600.4.10" "-"
84.57.178.67 - - [04/Apr/2015:22:28:57 +0200] "GET /wp-content/themes/getnoticed/inc/admin/images/getnoticed-logo.svg HTTP/1.1" 200 793 www.geemco.de "http://www.geemco.de/wp-admin/index.php" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10_2) AppleWebKit/600.4.10 (KHTML, like Gecko) Version/8.0.4 Safari/600.4.10" "-"
84.57.178.67 - - [04/Apr/2015:22:28:57 +0200] "POST /wp-admin/admin-ajax.php HTTP/1.1" 200 17 www.geemco.de "http://www.geemco.de/wp-admin/index.php" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10_2) AppleWebKit/600.4.10 (KHTML, like Gecko) Version/8.0.4 Safari/600.4.10" "-"
84.57.178.67 - - [04/Apr/2015:22:28:57 +0200] "GET /wp-admin/admin-ajax.php?action=blc_dashboard_status&random=0.746795387705788 HTTP/1.1" 200 306 www.geemco.de "http://www.geemco.de/wp-admin/index.php" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10_2) AppleWebKit/600.4.10 (KHTML, like Gecko) Version/8.0.4 Safari/600.4.10" "-"
84.57.178.67 - - [04/Apr/2015:22:29:08 +0200] "GET /wp-admin/plugins.php HTTP/1.1" 200 115975 www.geemco.de "http://www.geemco.de/wp-admin/index.php" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10_2) AppleWebKit/600.4.10 (KHTML, like Gecko) Version/8.0.4 Safari/600.4.10" "-"
84.57.178.67 - - [04/Apr/2015:22:29:10 +0200] "POST /wp-admin/admin-ajax.php HTTP/1.1" 200 17 www.geemco.de "http://www.geemco.de/wp-admin/plugins.php" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10_2) AppleWebKit/600.4.10 (KHTML, like Gecko) Version/8.0.4 Safari/600.4.10" "-"
84.57.178.67 - - [04/Apr/2015:22:30:11 +0200] "POST /wp-admin/admin-ajax.php HTTP/1.1" 200 47 www.geemco.de "http://www.geemco.de/wp-admin/plugins.php" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10_2) AppleWebKit/600.4.10 (KHTML, like Gecko) Version/8.0.4 Safari/600.4.10" "-"
84.57.178.67 - - [04/Apr/2015:22:31:13 +0200] "POST /wp-admin/admin-ajax.php HTTP/1.1" 200 47 www.geemco.de "http://www.geemco.de/wp-admin/plugins.php" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10_2) AppleWebKit/600.4.10 (KHTML, like Gecko) Version/8.0.4 Safari/600.4.10" "-"
84.57.178.67 - - [04/Apr/2015:22:31:20 +0200] "GET /wp-admin/update.php?action=upgrade-plugin&plugin=podlove-podcasting-plugin-for-wordpress%2Fpodlove.php&_wpnonce=5b9b85dca8 HTTP/1.1" 200 65739 www.geemco.de "http://www.geemco.de/wp-admin/plugins.php" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10_2) AppleWebKit/600.4.10 (KHTML, like Gecko) Version/8.0.4 Safari/600.4.10" "-"
84.57.178.67 - - [04/Apr/2015:22:31:53 +0200] "GET /wp-admin/plugins.php HTTP/1.1" 200 115975 www.geemco.de "http://www.geemco.de/wp-admin/update.php?action=upgrade-plugin&plugin=podlove-podcasting-plugin-for-wordpress%2Fpodlove.php&_wpnonce=5b9b85dca8" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10_2) AppleWebKit/600.4.10 (KHTML, like Gecko) Version/8.0.4 Safari/600.4.10" "-"
84.57.178.67 - - [04/Apr/2015:22:31:54 +0200] "POST /wp-admin/admin-ajax.php HTTP/1.1" 200 17 www.geemco.de "http://www.geemco.de/wp-admin/plugins.php" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10_2) AppleWebKit/600.4.10 (KHTML, like Gecko) Version/8.0.4 Safari/600.4.10" "-"

**Comment by timpritlove** [comment]
That's the access log, not the error log.

**Comment by GoetzMueller** [comment]
Gotcha ;-)
Twice out of memory, perfectly aligned with the update attempts:
[04-Apr-2015 20:31:29 UTC] PHP Fatal error:  Out of memory (allocated 34340864) (tried to allocate 8001522 bytes) in /homepages/23/d290384024/htdocs/wpgeemco/wp-admin/includes/file.php on line 646
[04-Apr-2015 21:54:39 UTC] PHP Fatal error:  Out of memory (allocated 34340864) (tried to allocate 8001522 bytes) in /homepages/23/d290384024/htdocs/wpgeemco/wp-admin/includes/file.php on line 646

**Comment by eteubert** [comment]
> Could the local changes that @eteubert made with the plugin code in my installation be of any influence?

No, local changes are simply overwritten when updating.

> Could anything go wrong when I upload the 2.0.5 code manually via FTP?

Not if you use the download from WordPress https://wordpress.org/plugins/podlove-podcasting-plugin-for-wordpress/developers/

> Twice out of memory, perfectly aligned with the update attempts

32M is very low, I am surprised you didn't run into issues earlier. Try http://codex.wordpress.org/Editing_wp-config.php#Increasing_memory_allocated_to_PHP
The error itself is only helpful if you can enable more details. But that requires enabling a PHP module (xdebug) and that's a bit fiddly. Increasing the memory should solve your problem though, so never mind :)

**Comment by GoetzMueller** [comment]
OK, I did the upload manually via FTP and it seems to be ok.

What I've noticed is growth in size of the plugin from 12 MB to 35 MB which is almost triple.

This is mainly caused by a subdirectory named "bower_components" with 23 MB containing two subdirectories "dist" with 14.3 MB (with subdirs "css" w/ 4 MB, "examples" w/ 8.7 MB (ultimately a file "Kapitel-Demo.m4a" w/ 8 MB), "js" w/ 1.3 MB) and "src" with 8.9 MB (with subdir "examples" looking pretty much the same as the subdir "examples" above, specially the "Kapitel-Demo.m4a" file).

This looks quite astonishing to me for a minor update from 2.0.2 to 2.0.5. Could it be that some of these files slipped somehow into the distribution zip file?

Apart of the that, the issue can be closed if the growth in size is of no further relevance.
